test/conftest.py

```diff
@@ -61,6 +61,41 @@ def pytest_addoption(parser):
     )
 
 
+@pytest.hookimpl(hookwrapper=True)
+def pytest_runtest_makereport(item, call):
+    """Make timeouts appear as such in the logs.
+
+    The pytest-timeout plug-in messes with the durations that are reported for tests.
+    Fix that by marking the report such that we can print `TIMEOUT` instead of
+    `FAILED (incorrect duration)`. See
+    https://github.com/pytest-dev/pytest/blob/38d8deb74d95077ebf189440ca047e14f8197da1/src/_pytest/runner.py#L366
+    """
+    r = yield
+    if not hasattr(r, 'get_result'):
+        return
+    report = r.get_result()
+    timed_out = False
+    if hasattr(call.excinfo, 'value'):
+        msg = getattr(call.excinfo.value, 'msg', None)
+        if isinstance(msg, str) and msg.startswith('Timeout >'):
+            timed_out = True
+    report.timed_out = timed_out
+
+
+def pytest_report_teststatus(report, **kwargs):
+    """Report test run duration, adapted from
+    https://github.com/pytest-dev/pytest/blob/38d8deb74d95077ebf189440ca047e14f8197da1/src/_pytest/runner.py#L202
+    """
+    d = report.duration
```

Comment:
Although all CI tests completed successfully, another
```py
if not hasattr(report, 'duration'):
    return None
```
might be necessary here?

Reply:
Why?

Reply:
`pypy3` failed locally while collecting tests. I tried to fix it using the additional check, now it's complaining about `pickle` that there is no protocol 5. `py39` collecting phase is okay, but the second test doesn't even start and doesn't time out neither. Something is badly wrong on my system. But that's not the point. The point is that I once saw an `AttributeError` (`CollectionReport` has no ... or something?) and this shouldn't cause any trouble. It's really weird, because I never had anything like this previously, even without the proposed check...

Reply:
Recreating the `tox -e pypy3` environment fixed that `pickle` error, but the second test doesn't start (same behaviour as with `tox -e py39`). This is also true for `upstream/master` and `upstream/master^`. Have to find out what's wrong with my setup.